src/pip/_internal/commands/install.py

```diff
@@ -732,3 +737,25 @@ def create_os_error_message(error, show_traceback, using_user_site):
         parts.append(".\n")
 
     return "".join(parts).strip() + "\n"
+
+
+def prepare_sudo_warning():
+    # type: () -> None
+    """Output a warning for sudo users on Unix.
+
+    On Windows, users may run pip as Administrator without issues.
+    """
+    try:
+        if sys.platform == 'darwin' or sys.platform == 'linux':
+            warn_sudo = os.getuid() == 0
+        elif sys.platform == 'win32' or sys.platform == 'cygwin':
+            warn_sudo = False
+    except AttributeError:
+        warn_sudo = False
+
+    if warn_sudo and not running_under_virtualenv():
+        sudo_warning = ("Running pip as root is not recommended! "
+                        "This could result in the corruption of "
+                        "package directories when a package manager is "
+                        "used. Consider using --user instead.")
+        logger.info(sudo_warning)
```

Comment:
If it's a warning, it should definitely be a `logger.warning`.

Reply:
I think this could be either an info or a warning. But I didn't use `logger.warning` because it ends up failing most of the test cases...the tests are run with sudo, but don't like when warnings appear.

Is there an elegant way around this (short of adding exemptions for every test case)?

Reply:
The most simplistic solution would be to monkeypatch `warn_if_run_as_root` to `lambda: None`.

Reply:
Changed to logger.warning. No monkeypatch needed.